Bash-Release: 3.2
Patch-ID: bash32-001

Bug-Reported-by:    Greg Schafer <gschafer@zip.com.au>
Bug-Reference-ID:   <20061012084940.GA15768@tigers.local>
Bug-Reference-URL: http://lists.gnu.org/archive/html/bug-bash/2006-10/msg00046.html

Bug-Description:

When using historical ``-style command substitution, bash incorrectly attempts
to interpret shell comments while scanning for the closing backquote.

--- bash-3.2/parse.y.ark	2006-09-19 22:37:21.000000000 +0200
+++ bash-3.2/parse.y	2006-10-17 16:53:20.000000000 +0200
@@ -2735,7 +2735,7 @@
 /* itrace("parse_matched_pair: open = %c close = %c", open, close); */
   count = 1;
   pass_next_character = backq_backslash = was_dollar = in_comment = 0;
-  check_comment = (flags & P_COMMAND) && qc != '\'' && qc != '"' && (flags & P_DQUOTE) == 0;
+  check_comment = (flags & P_COMMAND) && qc != '\'' && qc != '"' && qc != '`' && (flags & P_DQUOTE) == 0;
 
   /* RFLAGS is the set of flags we want to pass to recursive calls. */
   rflags = (qc == '"') ? P_DQUOTE : (flags & P_DQUOTE);
